EventPlanning.BL/InvitationManager.cs

```diff
@@ -0,0 +1,146 @@
+
+            return false;
+        }
+
+        /// <summary>
+        /// Decides what kind of invitation to send
+        /// </summary>
+        /// <param name="eventId"></param>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public string SendInvitation(int eventId, UserData user)
+        {
+            if (user.Email != null && SendMail(eventId, user))
+            {
+                return "Email";
+            }
+
+            if (user.Id != 0 && SendNotification(eventId, user))
+            {
+                return "Notification";
+            }
+
+            return "";
+        }
+
+        public bool SendNotification(int eventId, UserData user)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Creates invitation
+        /// </summary>
+        /// <param name="eventId"></param>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public bool CreateInvitation(int eventId, UserData user, string sentTo)
+        {
+            if(sentTo != "Email" && sentTo != "Notification")
+            {
+                return false;
+            }
+
+            var invitation = new Invitation()
+            {
+                EventId = eventId,
+                SentTo = sentTo,
+                DateSent = DateTime.Now,
+                UserId = user.Id,
+                Status = "Waiting",
+            };
+
+            return repository.AddInvitation(invitation);
+        }
+
+        /// <summary>
+        /// Sends email with invitation
+        /// </summary>
+        /// <param name="eventId"></param>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public bool SendMail(int eventId, UserData user)
+        {
+            var ev = repository.GetEvent(eventId);
+            var mailBody = new StringBuilder();
+            mailBody.AppendFormat("You have been invited to event named: {0}.", ev.Title).AppendLine();
+            mailBody.Append("To confirm participation you must click link below.").AppendLine();
+
+            var acceptLink = Environment.GetEnvironmentVariable("acceptLink");
+
+            if (acceptLink == null)
+            {
+                return false;
+            }
+
+            mailBody.AppendFormat("{0}/{1}/{2}",acceptLink, eventId, user.Id);
+
+            MailMessage mail = new MailMessage();
+            mail.From = new MailAddress("eventinvitator@gmail.com");
+            mail.Subject = "Invitation to event";
+            mail.Body = mailBody.ToString();
+
+            if (user.Email == null || user.Email == "")
+            {
+                return false;
+            }
+
+            try
+            {
+                mail.To.Add(user.Email);
+                smtpServer.Send(mail);
+            }
+            catch (FormatException)
+            {
+                return false;
+            }
+            catch (SmtpFailedRecipientsException)
```

Comment:
Let's try using exception filters 
https://stackoverflow.com/questions/5563397/catch-two-exceptions-in-the-same-catch-block